examples/example_endpoints.py

```diff
@@ -142,3 +153,84 @@ def predict(
 
     print(answers)
     # TODO what type answers are?
+
+
+@persistent_cache('evaluate')
+def evaluate(
+    dataset_id: Optional[UUID] = None,
+    dataset_name: Optional[str] = None,
+    retriever_type: str = 'dpr',
+    dpr_question_encoder: str = 'facebook/dpr-question_encoder-single-nq-base',
+    dpr_context_encoder: str = 'facebook/dpr-ctx_encoder-single-nq-base',
+    reader_encoder: str = 'deepset/roberta-base-squad2',
+):
+
+    # todo check if results are already in the evaluation cache
+
+
+    if dataset_id is not None:
+        dataset = dataset_repository.get(dataset_id)
+    elif dataset_name is not None:
+        dataset_downloader = DatasetDownloader()
+        if dataset_name == DatasetDownloader.NQ_KEY or dataset_name == 'natural_questions':
+            dataset_path = dataset_downloader.download(DatasetDownloader.NQ_KEY)
+        elif dataset_name == DatasetDownloader.SQUAD_KEY:
+            dataset_path = dataset_downloader.download(DatasetDownloader.SQUAD_KEY)
+        else:
+            logger.warning(f"No such dataset key as '{dataset_name}'")
+            raise ValueError(f"No such dataset key as '{dataset_name}'")
+
+        original_docs, _ = eval_data_from_json(filename=str(dataset_path),
+                                               max_docs=None,
+                                               preprocessor=None)
+
+        split_preprocessor = PreProcessor(split_by='word',
+                                          split_length=200,
+                                          split_overlap=0,
+                                          split_respect_sentence_boundary=False,
+                                          clean_empty_lines=False,
+                                          clean_whitespace=False)
+
+        preprocessed_docs, labels = eval_data_from_json(filename=str(dataset_path),
+                                                        max_docs=None,
+                                                        preprocessor=split_preprocessor)
+        try:
+            # check if already exists with such name - if not -> add uuid
+            corpus = corpus_repository.get_by_name(dataset_name)  # is it working like it should?
+            if corpus is not None:
+                dataset_name += '_'
+                dataset_name += str(uuid.uuid4())
+
+            # Czy .add() i .commit() powinny być w tym miejscu, czy jakoś na końcu?
+
+            corpus = DataCorpus(name=dataset_name)
+            corpus_repository.add(corpus)
+            corpus_repository.commit()
+
+            dataset = Dataset(name=dataset_name, corpus=corpus)
+            dataset_repository.add(dataset)
+            dataset_repository.commit()
+
+            ELASTICSEARCH_STORAGE.add_dataset(preprocessed_docs=preprocessed_docs,
```

Comment:
Nie wiem czy nie warto by to wszystko (podział, tworzenie obiektów Document i td) robiła ta funkcja. Trochę to nie odpowiada tej metodzie z dodaniem dokumentu i trochę samo złe jest, bo właśnie to mi się nie podobało w Haystack, że wymagają ścieżki, a nie samych obiektów, i nie jest to rozdzielone, ale warto będzie jeszcze nad tym pomyślić, bo za dużo różnych rzeczy się przekazuje do tej metody

Reply:
No trochę dziwnie to teraz wygląda, ale no to przez to że dokumenty są haystackowe a nie nasze